uproot/write/objects/TTree.py

```diff
@@ -73,6 +99,9 @@ def _write(self, context, cursor, name, compression, key, keycursor, util):
         self._tree.write(context, cursor, name, key, copy(keycursor), util)
 
     def extend(self, branchdict):
+
+        #FIXME: Enforce that the inner array lengths of multiple jagged array having the same dependence are the same
```

Comment:
This is important and not too hard to check. Just verify that the branches in `branchdict` with the same dependence have the same `jaggedarray.counts` (by `(test1.counts == test2.counts).all()`).

Reply:
This should be fixed now.